### DISCUSSION TITLE
What is the difference between stream and device?

### DISCUSSION THREAD

**Discussion by Zhiming-Huang** [ml-explore/mlx#1233] [Q&A]
MLX support to set a stream or a device for each function. What is the relationship between Stream and Device?

**Comment by awni** [comment]
Every stream is associated with a device. A device can have one or more streams. An operation can take as input a stream or a device. If you give it a stream then the operation runs on that stream. If you give it a device, then it runs on the default stream of the default device.

For more inofrmation checkout the documenation:

https://ml-explore.github.io/mlx/build/html/usage/using_streams.html
https://ml-explore.github.io/mlx/build/html/python/devices_and_streams.html
